src/LibraryViewExtension/Handlers/NodeItemDataProvider.cs

```diff
@@ -48,16 +52,13 @@ public override Stream GetResource(IRequest request, out string extension)
             return GetNodeItemDataStream(model.SearchEntries);
         }
 
-        private Stream GetNodeItemDataStream(IEnumerable<NodeSearchElement> searchEntries)
+        protected Stream GetNodeItemDataStream(IEnumerable<NodeSearchElement> searchEntries)
         {
-            var data = new LoadedTypeData();
-            data.loadedTypes = searchEntries
-                //.Where(e => !e.ElementType.HasFlag(ElementTypes.Packaged))
-                .Select(e => CreateLoadedTypeItem(e)).ToList();
-
             var ms = new MemoryStream();
             var sw = new StreamWriter(ms);
             var serializer = new JsonSerializer();
+
+            var data = CreateObjectForSerialization(searchEntries);
```

Comment:
CreateObjectForSerialization basically creates different instance of LoadedTypeData in the base class and the derived class.